### DISCUSSION TITLE
Method Illuminate\Validation\Validator::validateAcceptedIf does not exist

### DISCUSSION THREAD

**Discussion by reasecret** [livewire/livewire#4325] [1. Help] [answered]
On Laravel 8 with Livewire I'm trying to use accepted_if rule but get this error:
Method Illuminate\Validation\Validator::validateAcceptedIf does not exist.

There is an agreement checkbox that must be accepted any condition. And there is another checkbox that must be accepted if contract column is not null on projects table. Contract column is nullable because some projects may not have contract.

If I use only **accepted** validation rule, projects that have no contract are not passing validation. I'm not sure this is Laravel or Livewire error. What should I do to fix this issue?

My blade file:
```
<div>
	<div class="inline-flex items-center space-x-2">
		<input wire:model.defer="agreement" id="agreement" type="checkbox">
		<label for="agreement">I agree to the terms and conditions</label>
		@error('agreement')
			<span class="text-xs text-red-500">{{ $message }}</span>
		@enderror
	</div>
</div>
@if($project->contract)
<div>
	<div class="inline-flex items-center space-x-2">
		<input wire:model.defer="contract" id="contract" type="checkbox">
		<label for="contract">I accept the<a href="#">contract</a></label>
		@error('contract')
			<span class="text-xs text-red-500">{{ $message }}</span>
		@enderror
	</div>
</div>
@endif
<input name="contractValue" type="hidden" value="{{ ($project->contract ? '1' : '0') }}">
```

My component:
```
	public function sendComment()
	{
		$this->validate([
			'agreement' => 'accepted',
			'contract' => 'accepted_if:contractValue,1',
		],
		[
			'agreement.accepted' => 'Agreement must accepted',
			'contract.accepted_if' => 'Contract must accepted',
		]);
	}
```

**Comment by joshhanley** [comment, marked as answer]
@reasecret what version of Laravel 8 are you using? `accepted_if` was released in v8.53.0, so maybe try updating your Laravel version and seeing if that works?

Hope this helps!

**Reply by reasecret** [reply]
Thanks @joshhanley. I thought accepted_if rule available all 8.x versions. There is no information about this on validation page. This error has gone after updating Laravel. 

But now there is another issue. contract checkbox is not validating with this rule: 'contract' => 'accepted_if:contractValue,1',

If checkbox is not checked there is no validation and no error message. What do you think about this?

**Reply by joshhanley** [reply]
@reasecret Laravel releases new features nearly every week, so trying to note what version it was released in would be hard to maintain. Generally always best to ensure you are using the latest version by updating regularly or updating if a documented feature isn't working 🙂

I'd have a look at what the value of `contractValue` is, you might find it is actually `true` instead of `1` possibly?

**Reply by reasecret** [reply]
contractValue comes from here:
`<input name="contractValue" type="hidden" value="{{ ($project->contract ? '1' : '0') }}">`

**Reply by joshhanley** [reply]
@reasecret ah Livewire doesn't work with hidden inputs, you should just set a property on the component with the data instead

**Reply by reasecret** [reply]
Ok got it. I use like this:

```
public $contractValue;

public function render()
    {
		if($this->project->contract){
			$this->contractValue = 1;
		}else{
			$this->contractValue = 0;
		}
        return view('livewire.single-project')->extends('layouts.app')->section('content');
    }

$this->validate([
		'contract' => 'accepted_if:contractValue,1',
		]);

```
Like this I don't need hidden input. Looks like no problem. Thanks again.

**Comment by devcircus** [comment]
accepted_if was added in Laravel 8.53. Are you on at least that version?

Thanks,

Clay
________________________________
From: Okan K ***@***.***>
Sent: Wednesday, December 1, 2021 2:15:12 PM
To: livewire/livewire ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [livewire/livewire] Method Illuminate\Validation\Validator::validateAcceptedIf does not exist (Discussion #4325)


On Laravel 8 with Livewire I'm trying to use accepted_if rule but get this error:
Method Illuminate\Validation\Validator::validateAcceptedIf does not exist.

There is an agreement checkbox that must be accepted any condition. And there is another checkbox that must be accepted if contract column is not null on projects table. Contract column is nullable because some projects may not have contract.

If I use only accepted validation rule, projects that have no contract are not passing validation. I'm not sure this is Laravel or Livewire error. What should I do to fix this issue?

My blade file:

<div>
        <div class="inline-flex items-center space-x-2">
                <input wire:model.defer="agreement" id="agreement" type="checkbox">
                <label for="agreement">I agree to the terms and conditions</label>
                @error('agreement')
                        <span class="text-xs text-red-500">{{ $message }}</span>
                @enderror
        </div>
</div>
@if($project->contract)
<div>
        <div class="inline-flex items-center space-x-2">
                <input wire:model.defer="contract" id="contract" type="checkbox">
                <label for="contract">I accept the<a href="#">contract</a></label>
                @error('contract')
                        <span class="text-xs text-red-500">{{ $message }}</span>
                @enderror
        </div>
</div>
@endif
<input name="contractValue" type="hidden" value="{{ ($project->contract ? '1' : '0') }}">


My component:

        public function sendComment()
        {
                $this->validate([
                        'agreement' => 'accepted',
                        'contract' => 'accepted_if:contractValue,1',
                ],
                [
                        'agreement.accepted' => 'Agreement must accepted',
                        'contract.accepted_if' => 'Contract must accepted',
                ]);
        }

**Reply by joshhanley** [reply]
@devcircus please see their answer above 🙂

**Reply by devcircus** [reply]
Ahh thanks. my phone email client had the answer hidden.
